src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCollectionDebuggerProxy.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace System.Text.RegularExpressions
+{
+    internal sealed class RegexCollectionDebuggerProxy<T>
```

Comment:
I think it would be reasonable for the EE to handle ICollection<T> automatically without the need for proxies, if it doesn't do so already. However, if the logic is not present in older EEs removing the proxy would degrade the experience (might be ok if it would only affects Dev10 and prior).

I think it is possibly to specify a proxy as an assembly-level attribute, which specifies the target type. Not sure if it is possible to target an interface though, probably not. BTW, I'm not positive but I think that the current debugger proxy architecture requires the target type to be serializable.